extensions/wikia/CreateNewWiki/tasks/SetCustomSettings.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Wikia\CreateNewWiki\Tasks;
+
+use User;
+use WikiFactory;
+use Wikia\Logger\Loggable;
+
+class SetCustomSettings implements Task {
+	use Loggable;
+
+	private $taskContext;
+
+	public function __construct( TaskContext $taskContext ) {
+		$this->taskContext = $taskContext;
+	}
+
+	public function prepare() {
+		return TaskResult::createForSuccess();
+	}
+
+	public function check() {
+		return TaskResult::createForSuccess();
+	}
+
+	public function run() {
+		global $wgUniversalCreationVariables, $wgLangCreationVariables;
+
+		$wikiType = "default";
+		$language = $this->taskContext->getLanguage();
+
+		if ( !empty($wgUniversalCreationVariables) && isset($wgUniversalCreationVariables[$wikiType]) ) {
+			$this->addCustomSettings( 0, $wgUniversalCreationVariables[$wikiType], "universal" );
+			$this->debug( implode( ":", [ "CreateWiki", __METHOD__, "Custom settings added for wiki_type: {$wikiType}" ] ) );
+		}
+
+		/**
+		 * set variables per language
+		 */
+		$langCreationVar = isset($wgLangCreationVariables[$wikiType]) ? $wgLangCreationVariables[$wikiType] : $wgLangCreationVariables;
+		$this->addCustomSettings( $language, $langCreationVar, "language" );
+		$this->debug( implode( ":",
+			[ "CreateWiki", __METHOD__, "Custom settings added for wiki_type: {$wikiType} and language: {$language}" ]
+		) );
+
+		return TaskResult::createForSuccess();
+	}
+
+	public function addCustomSettings( $match, $settings, $type = 'unknown' ) {
+		global $wgUser;
+
+		wfProfileIn( __METHOD__ );
+
+		if ( (!empty($match) || $type == 'universal') && isset($settings[$match]) && is_array( $settings[$match] ) ) {
+			$this->debug( implode( ":", [ "CreateWiki", __METHOD__, "Found '$match' in {$type} settings array" ] ) );
+
+			/**
+			 * switching user for correct logging
+			 */
+			$oldUser = $wgUser;
+			$wgUser = User::newFromName( 'CreateWiki script' );
```

Comment:
This is a very bald suggestion but we could just extend setVarById to have an optional argument with user instead of manipulating global variable ;) But I'll understand if we don't want to change wiki factory ATM.